src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -3,19 +3,58 @@
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using System.Net.Http;
     using System.Reflection;
+    using System.Threading.Tasks;
 
     internal static class HeartbeatDefaultPayload
     {
-       public static readonly string[] DefaultFields =
-        {
+        public static readonly string[] DefaultFields =
+         {
             "runtimeFramework",
-            "baseSdkTargetFramework"
+            "baseSdkTargetFramework",
+            "osVersion"
+        };
+
+        public static readonly string[] DefaultOptionalFields =
+         {
+            "processId",
+            // the  following are Azure Instance Metadata fields
+            "osType",
```

Comment:
for non-AzureVM also we need osType.

Reply:
Yes, I have that covered in the osVersion field. Further discussion of that below.